Core/Executor/UserManager.php

```diff
@@ -111,25 +115,29 @@ protected function update()
 
                 $assignedGroups = $userService->loadUserGroupsOfUser($user);
 
+                $targetGroupIds = [];
                 // Assigning new groups to the user
                 foreach ($this->dsl['groups'] as $groupToAssignId) {
+                    $groupId = $this->referenceResolver->resolveReference($groupToAssignId);
+                    $groupToAssign = $this->userGroupMatcher->matchOneByKey($groupId);
```

Comment:
I presume that this means you could identify the groups to assign to a user by ref as well as remote id, correct?